builder/build_validate.go

```diff
@@ -254,46 +263,81 @@ func (b *Builder) mcaPkgInfo(mInfo map[string]*swupd.Manifest, version, download
 	pInfo := make(map[string]*mcaBundlePkgInfo)
 
 	// Download and query file metadata from all packages in each bundle
+	bundleWorker := func() {
+		for m := range mCh {
+			var pkgList = []string{}
+			for pkg := range mInfo[m.name].BundleInfo.AllPackages {
+				pkgList = append(pkgList, pkg)
+			}
+
+			out, err := downloadRpms(packagerCmd, pkgList, buildVersionDir, downloadRetries)
+			if err != nil {
+				errorCh <- err
+				wg.Done()
+				return
+			}
+
+			// Collect metadata to resolve installed RPM file names
+			bundlePkgInfo := pkgInfoFromNoopInstall(out.String())
+			for _, p := range bundlePkgInfo {
+				rw.RLock()
+				name := pkgCache[p.name]
+				rw.RUnlock()
+
+				// Resolve files for package when it doesn't exist in the cache
+				if name == nil {
+					p.files, err = b.resolvePkgFiles(p, version)
+					if err != nil {
+						errorCh <- err
+						wg.Done()
+						return
+					}
+					rw.Lock()
```

Comment:
you still need to check if `pkgCache[p.name] == nil` here before writing since another goroutine could have filled this entry before this one has finished. This is not a big deal since the information should be identical, but I would at least add a comment here explaining why the override is not a TOCTOU error.

Reply:
Added this check